### ISSUE TITLE
Table of content always jump to the first header if multiple headers share same title.

### ISSUE THREAD

**Issue by ann2048** [KosmosisDire/obsidian-webpage-export#324]
In right-sidebar area (Table of contents), if multiple headers share the same title, no matter which title I clicked, the page always return to the first title having that name.

You can export following test file for example.

[Test_file.md](https://github.com/KosmosisDire/obsidian-webpage-export/files/14078824/Test_file.md)

**Comment by ann2048** [comment]
It seems the problem comes from duplicated ID from the header title. 
I added an index to "id" of header and "href" of table of content to solve it:

1. Add "index" inside `fixLinks()` function:
```js
  file.document.querySelectorAll("h1, h2, h3, h4, h5, h6").forEach((headerEl, index) => {
      var _a, _b, _c;
      headerEl.setAttribute("id", (_c = (_b = (_a = headerEl.getAttribute("data-heading")) != null ? _a : headerEl.textContent) == null ? void 0 : _b.replaceAll(" ", "_")) != null ? _c : "");
      let _id_ = headerEl.getAttribute("id");
      headerEl.setAttribute("id", _id_ + "_" + index);
    });
```

2. Add another function `fixLinksOutlineTree()` to fix table of contents:
```js
  static fixLinksOutlineTree(file) {
    if (!file.document)
      return;
    file.document.querySelectorAll(".tree-container.outline-tree .internal-link.tree-item-link").forEach((linkEl, index) => {
      let href = linkEl.getAttribute("href");
      linkEl.setAttribute("href", href + "_" + index);
    });
  }
``` 
3. Put fixLinksOutlineTree in side `create()` after `await headerTree.generateTreeWithContainer(rightSidebar);`:
```js
    if (MainSettings.settings.includeOutline) {
      let headerTree = new OutlineTree(this.source, 1);
      headerTree.class = "outline-tree";
      headerTree.title = "Table Of Contents";
      headerTree.showNestingIndicator = false;
      headerTree.generateWithItemsClosed = MainSettings.settings.startOutlineCollapsed;
      await headerTree.generateTreeWithContainer(rightSidebar);
      GenHelper.fixLinksOutlineTree(this);
    }
```

**Comment by KosmosisDire** [comment]
Thanks for the suggestion. The reason I haven't done this is because the header's ID is supposed to be the same as the header's text (this makes it nice when sharing the url `example.com#Header` with automatically scrol to that header). If two headers have identical text this causes them to be indistiguishable. And the second reason is because in standard html and markdown you're not reeeally supposed to ever have two headers with the same text. I could break some of those standards to allow it to work, but it would be a lot of work for a very small issue.

I may implement this eventually, but at the moment it's probably not high on the list. Thanks! :)

**Comment by KosmosisDire** [comment]
Does your solution only take into account the outline or does it also take raw links (even ones from other notes) and headings in the url into account as well?

**Comment by ann2048** [comment]
No, the solution can only solve the table of content. I currently have no idea how to modify all the internal links as well.

**Comment by ann2048** [comment]
How about adding its parents title to a child's id?
For example, if I have "projectA", "projectB", "projectC" as h1, and all of the headers have "Results" in h2, the "id" of h2 may be "projectA_Results", "projectB_Results", "projectC_Results". 
I will try it.

**Comment by KosmosisDire** [comment]
That could work as well, but you'll end up with very long ids. Remember we want to be able to have these at the end of the urls. So only duplicated header IDs should be different.

**Comment by KosmosisDire** [comment]
I think it would actually be better to not change the id's at all, because obsidian actually doesn't handle internal links to headers with the same name. Obsidian also just goes to the first one. I think a better solution is to handle the duplication on the frontend / client rather than in the generation code. Because like you implied the more important usage of this is to make the outline go to the correct header.

**Comment by ann2048** [comment]
I think you're right, it's better to avoid same title in headers. Thanks for your comments!

**Comment by KosmosisDire** [comment]
I will keep this open because I would like to add support for this to the outline on the frontend in a different way. But it, again, won't be super high on the list of priorities

Thanks! :)
